### PR TITLE
{Extension} Add retry when getting extension index

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**

We're seeing increasing failures of `Test Extensions Loading Python36` task in CI when downloading the extension index file. Add retry to mitigate the issue while we trying to figure out the root cause.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Extension

File: src/azure-cli-core/azure/cli/core/extension/_index.py [hunk 1]
```diff
@@ -33,8 +33,12 @@ def get_index(index_url=None):
...
             raise CLIError(msg)
         except (requests.exceptions.ConnectionError, requests.exceptions.HTTPError) as err:
-            msg = ERR_TMPL_NO_NETWORK.format(str(err))
-            raise CLIError(msg)
+            if try_number == TRIES - 1:
+                msg = ERR_TMPL_NO_NETWORK.format(str(err))
+                raise CLIError(msg)
+            import time
+            time.sleep(0.5)
+            continue
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/extension/_index.py:41]: We can combine `(requests.exceptions.ConnectionError, requests.exceptions.HTTPError)` and  `ValueError` as the handling code is the exactly the same.

Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/extension/_index.py]: make sense.

File: src/azure-cli-core/azure/cli/core/extension/_index.py [hunk 2]
```diff
@@ -32,13 +32,14 @@ def get_index(index_url=None):
...
-            # Indicates that url is not redirecting properly to intended index url, we stop retrying after TRIES calls
+        except (requests.exceptions.ConnectionError, requests.exceptions.HTTPError, ValueError) as err:
             if try_number == TRIES - 1:
-                msg = ERR_TMPL_BAD_JSON.format(str(err))
+                if isinstance(err, ValueError):
+                    # Indicates that url is not redirecting properly to intended index url, we stop retrying after
+                    # TRIES calls
+                    msg = ERR_TMPL_BAD_JSON.format(str(err))
+                else:
+                    msg = ERR_TMPL_NO_NETWORK.format(str(err))
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/extension/_index.py:42]:
```suggestion
                msg = ERR_TMPL_BAD_JSON.format(str(err)) if isinstance(err, ValueError) else ERR_TMPL_NO_NETWORK.format(str(err))
```

**Review by jiasli — APPROVED** [review]

**Pull request merged**